File: cli/node/src/application/entity-ids.ts
import type { MemGoConfig } from '../config/types.js';
/**
 * 优先使用显式实体 ID；未指定任何 ID 时读取配置默认值。
 * @param config - 当前配置。
 * @param opts - 本次操作选项。
 * @param opts.userId - 用户 ID。
 * @param opts.agentId - 代理 ID。
 * @param opts.appId - 应用 ID。
 * @param opts.runId - 会话 ID。
 * @returns 本次调用使用的实体 ID。
 */
export function resolveIds(
  config: MemGoConfig,
  opts: {
    userId?: string;
    agentId?: string;
    appId?: string;
    runId?: string;
  }
): { userId?: string; agentId?: string; appId?: string; runId?: string } {
  const hasExplicit = !!(opts.userId || opts.agentId || opts.appId || opts.runId);
  if (hasExplicit) {
    return {
      userId: opts.userId || undefined,
      agentId: opts.agentId || undefined,
      appId: opts.appId || undefined,
      runId: opts.runId || undefined,
    };
  }
  return {
    userId: config.defaults.userId || undefined,
    agentId: config.defaults.agentId || undefined,
    appId: config.defaults.appId || undefined,
    runId: config.defaults.runId || undefined,
  };
}

File: cli/python/src/memgo_cli/application/entity_ids.py
"""显式实体 ID 与配置默认范围解析。"""

from __future__ import annotations

from memgo_cli.config.models import MemGoConfig


def _resolve_ids(
    config: MemGoConfig,
    *,
    user_id: str | None,
    agent_id: str | None,
    app_id: str | None,
    run_id: str | None,
) -> dict[str, str | None]:
    """解析实体范围。

    提供任意显式 ID 时仅使用显式值；否则读取配置默认值，避免混合范围。
    """
    has_explicit = any([user_id, agent_id, app_id, run_id])
    if has_explicit:
        return {
            "user_id": user_id or None,
            "agent_id": agent_id or None,
            "app_id": app_id or None,
            "run_id": run_id or None,
        }
    return {
        "user_id": config.defaults.user_id or None,
        "agent_id": config.defaults.agent_id or None,
        "app_id": config.defaults.app_id or None,
        "run_id": config.defaults.run_id or None,
    }
